fixed 2 errors in gpg2john.c

sieger82 opened the pull request:
1) in debug mode Getc() gets invoked 1 too many times resulting in eof errors
2) in debug mode salt is not added to outputted hash.

solardiz commented on src/gpg2john.c:
```diff
@@ -820,8 +821,14 @@ string_to_key(void)
+			}
+		}
 		else
 			give(8, m_salt, sizeof(m_salt));
```
I think both instances of the `give` call should ideally be replaced with one just before the `if`. And no need to clutter the code with comments about past bugs, the differences between these functions are obvious enough not to require that.

solardiz commented on src/gpg2john.c:
```diff
@@ -836,8 +843,14 @@ string_to_key(void)
+			}
+		}
 		else
 			give(8, m_salt, sizeof(m_salt));
```
Ditto.

solardiz commented on src/gpg2john.c:
```diff
@@ -1138,8 +1151,12 @@ Symmetrically_Encrypted_and_MDC_Packet(int len, int first, int partial, char *ha
 		totlen = 0;
-		if (gpg_dbg) fprintf(stderr, "\tVer %d\n", Getc());
-		Getc(); // version (we only read this from the first packet. Not read from rest of the 'partial' packets.
+		//FIXED: when using debug mode Getc() was invoked one too many times resulting in unexpected end of file errors.
```
No need for the comment.

solardiz reviewed (changes requested):
I don't mind merging this as-is (the original code is very dirty, and this is fixing real bugs), but ideally you'd clean up these things I noted and force-push first. Thanks!

solardiz commented:
Also, let me suggest a replacement commit message:

```
gpg2john: Fix 2 debug mode bugs

1. Getc() was invoked 1 too many times resulting in EOF errors
2. Salt was not added to outputted hash
```

This separates the title from detail and avoids repetition. By using "was", it also makes it clear the described behavior is gone.

sieger82 commented:
tnx for the comments. makes total sense. will fix and get back

sieger82 commented:
I've made the changes you requested.

solardiz commented on src/gpg2john.c:
```diff
@@ -820,10 +821,12 @@ string_to_key(void)
+				fprintf(stderr, "%02x", m_salt[i]);
+			}
+		}
 		if (gpg_dbg) fprintf(stderr, "\n");
```
Oh, surely this `fprintf` should go into the `if` block above instead of checking the same condition again. I'm sorry I didn't notice this right away - it became more apparent after the previous cleanup.

solardiz commented on src/gpg2john.c:
```diff
@@ -836,10 +839,12 @@ string_to_key(void)
+				fprintf(stderr, "%02x", m_salt[i]);
+			}
+		}
 		if (gpg_dbg) fprintf(stderr, "\n");
```
Ditto.

solardiz reviewed (changes requested):
Sorry, noticed two more cleanups to make within these changes.

solardiz commented on src/gpg2john.c:
```diff
@@ -836,17 +839,17 @@ string_to_key(void)
+		int count, c = Getc();
+		count = (16 + (c & 15)) << ((c >> 4) + EXPBIAS);
+		if (gpg_dbg) fprintf(stderr, "\t\tCount - %d(coded count %d)\n", count, c);
+		m_count = count;
```
Please revert the non-requested changes here. Having the scope of those two variables limited actually made sense.

sieger82 replied:
sorry, pushed too fast...

sieger82 committed:
gpg2john: Fix 2 debug mode bugs

1. Getc() was invoked 1 too many times resulting in EOF errors
2. Salt was not added to outputted hash

solardiz commented:
I think we're almost done here - let's wait for CI and merge. Thank you!
